cluster-autoscaler/go.mod

```diff
@@ -1,6 +1,8 @@
 module k8s.io/autoscaler/cluster-autoscaler
 
-go 1.21
+go 1.21.3
```

Comment:
yes i used

Reply:
Ok, I looked into it, and the go command put 1.21.3 here because k8s modules we depend on used to specify it, and a module has to have at least as high go version specified here as all its dependencies. It looks like after https://github.com/kubernetes/kubernetes/pull/121808 the dependencies reverted back to only specifying 1.21.

Could you re-run the script with the newest kubernetes/kubernetes master tag (`v1.30.0-alpha.3`), and then manually downgrade the go directive here to just `1.21`? The go command that `update-vendor.sh` uses will not automatically downgrade it even if all of the dependencies downgraded, and there's no reason to keep this higher IIUC, CA shouldn't depend on any differences in Go behavior between go1.21 and go1.21.3 (and neither do any of its dependencies on k8s `v1.30.0-alpha.3` if I'm looking correctly).

The toolchain directive should actually be good to specify, to the same go version that we're setting in the builder. This is only a suggestion for which go version to use while building, not a requirement for Go language features like the go directive.

Context: https://go.dev/doc/toolchain

Reply:
done @towca PTAL

Reply:
LGTM, thanks!